File: Examples/VisualBasic/ModifyingAndConvertingImages/JPEG/CroppingByRectangle.vb
Imports Aspose.Imaging

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.JPEG
    Class CroppingByRectangle
        Public Shared Sub Run()
            ' ExStart:CroppingByRectangle
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_JPEG()

            ' Load an existing image into an instance of RasterImage class
            Using rasterImage As RasterImage = DirectCast(Image.Load(dataDir & Convert.ToString("aspose-logo.jpg")), RasterImage)
                If Not rasterImage.IsCached Then
                    rasterImage.CacheData()
                End If

                ' Create an instance of Rectangle class with desired size
                Dim rectangle As New Rectangle(20, 20, 20, 20)

                ' Perform the crop operation on object of Rectangle class and Save the results to disk
                rasterImage.Crop(rectangle)
                rasterImage.Save(dataDir & Convert.ToString("CroppingByRectangle_out.jpg"))
            End Using
            ' ExEnd:CroppingByRectangle
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/JPEG/CroppingByRectangle.cs
// GIST-ID: 2c872364943e3b2f2256414328a6da93
/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions 
please feel free to contact us using https://forum.aspose.com/
*/

using System;
namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.JPEG
{
    class CroppingByRectangle
    {
        public static void Run()
        {
            //ExStart:CroppingByRectangle
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_JPEG();

            Console.WriteLine("Running example CroppingByRectangle");
            // Load an existing image into an instance of the RasterImage class.
            using (RasterImage rasterImage = (RasterImage)Image.Load(dataDir + "aspose-logo.jpg"))
            {
                if (!rasterImage.IsCached)
                {
                    rasterImage.CacheData();
                }

                // Create an instance of the Rectangle class with the desired size, perform the crop operation, and save the results to disk.
                Rectangle rectangle = new Rectangle(20, 20, 20, 20);
                rasterImage.Crop(rectangle);
                rasterImage.Save(dataDir + "CroppingByRectangle_out.jpg");
            }

            Console.WriteLine("Finished example CroppingByRectangle");
            //ExEnd:CroppingByRectangle
        }        
    }
}